modal/container_process.py

```diff
@@ -93,6 +101,10 @@ async def poll(self) -> Optional[int]:
         if self._returncode is not None:
             return self._returncode
 
+        if self._timeout and datetime.now() >= self._exec_start + self._timeout:
```

Comment:
Using `datetime `is likely not valid, because this clock can go backwards. I think you want https://docs.python.org/3/library/time.html#time.monotonic